corehq/apps/locations/tests/test_location_fixtures.py

```diff
@@ -849,3 +854,47 @@ def test_sync_format_with_disabled_toggle(self):
 
         self.assertEqual(False, should_sync_hierarchical_fixture(self.domain, app=None))
         self.assertEqual(True, should_sync_flat_fixture(self.domain, app=None))
+
+    @flag_enabled('HIERARCHICAL_LOCATION_FIXTURE')
+    def test_sync_format_with_app_aware_project_default(self):
+        app = MockApp(location_fixture_restore='project_default')
+        conf = LocationFixtureConfiguration.for_domain(self.domain.name)
+        conf.sync_hierarchical_fixture = True
+        conf.sync_flat_fixture = False
+        conf.save()
+
+        self.assertTrue(should_sync_hierarchical_fixture(self.domain, app))
+        self.assertFalse(should_sync_flat_fixture(self.domain, app))
+
+    @flag_enabled('HIERARCHICAL_LOCATION_FIXTURE')
+    def test_sync_format_with_app_aware_both_fixtures(self):
+        app = MockApp(location_fixture_restore='both_fixtures')
+        conf = LocationFixtureConfiguration.for_domain(self.domain.name)
+        conf.sync_hierarchical_fixture = False
+        conf.sync_flat_fixture = False
+        conf.save()
+
+        self.assertTrue(should_sync_hierarchical_fixture(self.domain, app))
+        self.assertTrue(should_sync_flat_fixture(self.domain, app))
+
+    @flag_enabled('HIERARCHICAL_LOCATION_FIXTURE')
+    def test_sync_format_with_app_aware_flat_fixtures(self):
+        app = MockApp(location_fixture_restore='only_flat_fixture')
+        conf = LocationFixtureConfiguration.for_domain(self.domain.name)
+        conf.sync_hierarchical_fixture = True
+        conf.sync_flat_fixture = False
+        conf.save()
+
+        self.assertFalse(should_sync_hierarchical_fixture(self.domain, app))
+        self.assertTrue(should_sync_flat_fixture(self.domain, app))
+
+    @flag_enabled('HIERARCHICAL_LOCATION_FIXTURE')
+    def test_sync_format_with_app_aware_hierarchical_fixture(self):
+        app = MockApp(location_fixture_restore='only_hierarchical_fixture')
+        conf = LocationFixtureConfiguration.for_domain(self.domain.name)
+        conf.sync_hierarchical_fixture = False
+        conf.sync_flat_fixture = True
+        conf.save()
+
+        self.assertTrue(should_sync_hierarchical_fixture(self.domain, app))
+        self.assertFalse(should_sync_flat_fixture(self.domain, app))
```

Comment:
These would also work well as a generator test

Reply:
thanks. i had thought of this, but didn't realize that you could use that with class tests https://github.com/dimagi/commcare-hq/pull/21664/commits/ed8b295dfa6825183ba4ec0c17cb6d3baa5aaf08